src/Hunk/SplitHunk/SplitChange.js

```diff
@@ -78,140 +78,132 @@ const renderCells = args => {
     ];
 };
 
-export default class SplitChange extends PureComponent {
-
-    static propTypes = {
-        oldSelected: PropTypes.bool.isRequired,
-        newSelected: PropTypes.bool.isRequired,
-        oldTokens: PropTypes.arrayOf(PropTypes.object),
-        newTokens: PropTypes.arrayOf(PropTypes.object),
-    };
-
-    static defaultProps = {
-        oldTokens: null,
-        newTokens: null,
-    };
-
-    state = {
-        hover: '',
-    };
-
-    constructor(props) {
-        super(props);
-
-        const unmarkHover = () => this.setState({hover: ''});
-        const markHoverAs = side => () => this.setState({hover: side});
-        const ownEventsOnOldSide = {
-            onMouseEnter: markHoverAs('old'),
+const SplitChange = props => {
+    const {
+        className,
+        gutterClassName,
+        codeClassName,
+        gutterEvents,
+        codeEvents,
+        oldChange,
+        newChange,
+        oldSelected,
+        newSelected,
+        oldTokens,
+        newTokens,
+        monotonous,
+        hideGutter,
+        generateAnchorID,
+        gutterAnchor,
+        renderToken,
+    } = props;
+
+    const [hover, setHover] = useState('');
+    const unmarkHover = useCallback(() => setHover(''), []);
+    const useCallbackOnSide = (side, customCallbacks) => {
+        const markHover = useCallback(() => setHover(side), []);
+        const hoverCallbacks = {
+            onMouseEnter: markHover,
             onMouseLeave: unmarkHover,
         };
-        const ownEventsOnNewSide = {
-            onMouseEnter: markHoverAs('new'),
-            onMouseLeave: unmarkHover,
+        const arg = {
+            side,
+            change: side === 'old' ? oldChange : newChange,
         };
+        // Unlike selectors, hooks do not provide native functionality to customize comparator,
+        // on realizing that this does not reduce amount of renders, only preventing duplicate merge computations,
+        // we decide not to optimize this extremely, leave it recomputed on certain rerenders.
+        const callbacks = useMemo(
+            () => mergeCallbacks(hoverCallbacks, customCallbacks, arg),
+            [unmarkHover, markHover, customCallbacks, arg.change]
```

Comment:
`unmarkHover, markHover, customCallbacks` will never be changed as `setHover`.

The only thing may be changed is `oldChange` or `newChange` in props.

Will this help?

Reply:
Exactly, `useCallback` unlike `useMemo` will never "forget" its memoized result, this can be simplified and may move outside the component function